libs/vkd3d/state.c

```diff
@@ -2332,6 +2333,11 @@ static uint32_t d3d12_graphics_pipeline_state_init_dynamic_state(const struct d3
         }
     }
 
+    /* We always need to enable fragment shading rate dynamic state
+     * D3D12 has no information about this ahead of time for a pipeline
+     * unlike Vulkan */
+    dynamic_state_flags |= VKD3D_DYNAMIC_STATE_FRAGMENT_SHADING_RATE;
```

Comment:
Shouldn't this be conditional on VRS being supported?

Reply:
:+1:

Reply:
Replaced with:

```c
    /* We always need to enable fragment shading rate dynamic state when rasterizing.
     * D3D12 has no information about this ahead of time for a pipeline
     * unlike Vulkan */
    if (d3d12_device_supports_variable_shading_rate_tier_1(state->device) && graphics->rt_count && graphics->sample_mask)
        dynamic_state_flags |= VKD3D_DYNAMIC_STATE_FRAGMENT_SHADING_RATE;```